stac_fastapi/types/stac_fastapi/types/rfc3339.py

```diff
@@ -45,53 +46,88 @@ def rfc3339_str_to_datetime(s: str) -> datetime:
+def parse_single_date(date_str: str) -> datetime:
+    """
+    Parse a single RFC3339 date string into a datetime object.
 
-    Interval strings are defined by
-    OGC API - Features Part 1 for the datetime query parameter value. These follow the
-    form '1985-04-12T23:20:50.52Z/1986-04-12T23:20:50.52Z', and allow either the start
-    or end (but not both) to be open-ended with '..' or ''.
+    Args:
+        date_str (str): A string representing the date in RFC3339 format.
+
+    Returns:
+        datetime: A datetime object parsed from the date_str.
+
+    Raises:
+        ValueError: If the date_str is empty or contains the placeholder '..'.
+    """
+    if ".." in date_str or not date_str:
+        raise ValueError("Invalid date format.")
+    return rfc3339_str_to_datetime(date_str)
+
+
+def validate_interval_format(values: list) -> None:
+    """
+    Validate the format of the interval string to ensure it contains at most
+    one forward slash.
 
     Args:
-        interval (str or None): The interval string to convert to a tuple of
-        datetime.datetime objects, or None if no datetime is specified.
+        values (list): A list of strings split by '/' from the interval string.
+
+    Raises:
+        HTTPException: If the interval string contains more than one forward slash.
+    """
+    if len(values) > 2:
+        raise HTTPException(
+            status_code=400,
+            detail="Interval string contains more than one forward slash.",
+        )
+
+
+def str_to_interval(interval: Optional[str]) -> Optional[DateTimeType]:
+    """
+    Extract a tuple of datetime objects from an interval string defined by the OGC API.
+    The interval can either be a single datetime or a range with start and end datetime.
+
+    Args:
+        interval (Optional[str]): The interval string to convert to datetime objects,
+        or None if no datetime is specified.
 
     Returns:
-        Optional[DateTimeType]: A tuple of datetime.datetime objects or None if
-        input is None.
+        Optional[DateTimeType]: A tuple of datetime.datetime objects or
+        None if input is None.
 
     Raises:
-        ValueError: If the string is not a valid interval string and not None.
+        HTTPException: If the string is not valid for various reasons such as being empty,
+        having more than one slash, or if date formats are invalid.
     """
     if interval is None:
         return None
 
     if not interval:
-        raise ValueError("Empty interval string is invalid.")
+        raise HTTPException(status_code=400, detail="Empty interval string is invalid.")
 
     values = interval.split("/")
-    if len(values) == 1:
-        # Single date for == date case
-        return rfc3339_str_to_datetime(values[0])
-    elif len(values) > 2:
-        raise ValueError(
-            f"Interval string '{interval}' contains more than one forward slash."
+    validate_interval_format(values)
```

Comment:
kinda weird to me to create a function just for this

Reply:
It's true - I was getting a complication error before with linting - I think I went too far in breaking things up

Reply:
just add `#noqa: C901` on top of the function to ignore those linting issue

Reply:
ok